Auth landing page

jennifer-lu opened the pull request:
## Notion ticket link
<!-- Please replace with your ticket's URL -->
[\[Auth\] Implement user sign on to Landing Page](https://www.notion.so/uwblueprintexecs/Auth-Implement-user-sign-on-to-Landing-Page-88ac6787cc224ff487340c23528ce8b7)


<!-- Give a quick summary of the implementation details, provide design justifications if necessary -->
## Implementation description
* Use `Auth.currentAuthenticatedUser()` to get authenticated user information
* Display user's name if authenticated


<!-- What should the reviewer do to verify your changes? Describe expected results and include screenshots when appropriate -->
## Steps to test
1. Go to http://localhost:3000/
2. Observe that header says "WELCOME TO YOUR TOOLKIT"
3. Go to http://localhost:3000/login
4. Log in, or sign up and log in (note the name you use)
5. Observe redirect to http://localhost:3000/
6. Observe that header says "WELCOME TO YOUR TOOLKIT, \[NAME\]"
7. Please also take a look at [documentation](https://www.notion.so/uwblueprintexecs/Authentication-455747d7459f4e969c768837621aa5bb) (see User section)


<!-- Draw attention to the substantial parts of your PR or anything you'd like a second opinion on
For Frontend changes be sure to specify which routes/pages reviewers should check out!
 -->
## What should reviewers focus on?
* Does it work?
* Does implementation make sense?


## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

jennifer-lu commented:
I added the `amplify` directory to `eslintignore` because linting was failing. Please let me know if this is an ok solution or if I should use a different fix.

ChinemeremChigbo commented:
> I added the `amplify` directory to `eslintignore` because linting was failing. Please let me know if this is an ok solution or if I should use a different fix.

For me running yarn lint --fix worked

- I removed eslint from .eslint ignore
![image](https://user-images.githubusercontent.com/24593519/156947508-f6fe3062-5be7-4403-bc4d-7fa35037c212.png)

- changed the single quotes in package.json to escaped double quotes as per ([why single quotes to escaped double quotes?](https://stackoverflow.com/questions/54543063/how-can-i-suppress-the-no-files-matching-the-pattern-message-in-eslint))
![image](https://user-images.githubusercontent.com/24593519/156947539-f57f86a5-9732-4bf8-93b9-e5f4ec5e87c5.png)

and then ran yarn lint --fix from the project's root directory

jennifer-lu commented:
> For me running yarn lint --fix worked
>
> ...
>
> and then ran yarn lint --fix from the project's root directory
>
> and everything still seems to run as usual:

@ChinemeremChigbo  Awesome, I'll make those changes. Thanks so much for the solution!

ChinemeremChigbo reviewed (approved):
Sweet! Looks great, thanks for making those changes.

chamod-gamage reviewed (changes requested):
Great work!
Not sure if this is in scope but the password requirement string has a typo (should be `lowercase`) and also can we make it wrap such that it doesn't increase the width of the input above it (the UX is kind of jarring when it makes the input super wide).
<img width="535" alt="image" src="https://user-images.githubusercontent.com/41309709/157780608-b5fd9592-bc4a-4cd1-8220-65d99823ae88.png">
There's also [some issues with the demographics form](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=6295f60944244182923b0160acd3cee6) but I believe that's out of scope for this PR.

jennifer-lu commented:
> Great work! Not sure if this is in scope but the password requirement string has a typo (should be `lowercase`) and also can we make it wrap such that it doesn't increase the width of the input above it (the UX is kind of jarring when it makes the input super wide).

Thanks for the review! I agree that the login UI should be polished, but I don't think it's in the scope of this ticket.

tawnyzhao commented on pages/index.tsx:
```diff
@@ -192,10 +193,22 @@ const Tools: React.FC<{ variant: string }> = ({ variant }) => {
         : "desktop";
 
+    useEffect(() => {
+        Auth.currentAuthenticatedUser()
```
You can wrap the asynchronous code inside a use effect in an async function to use async await syntax

jennifer-lu commented:
@tawnyzhao I've fixed the gitignore and removed the amplify files. Also changed the `useEffect` code. Please double check if you have time.

@chamod-gamage I also think that the login UI can be polished, but it isn't in the scope of this ticket. The fixes that you mentioned can be made in another pr (would appreciate it if you unrequest changes so I can merge).

chamod-gamage reviewed (approved):
Awesome work!
